packages/pxweb2/src/mappers/JsonStat2ResponseMapper.ts

```diff
@@ -145,17 +158,71 @@ function mapDimension(id: string, dimension: any, role: any): Variable | null {
     }
   }
 
+  const codeLists: Array<CodeList> = []; // Default empty array
+
   const variable: Variable = {
     id: id,
     label: dimension.label,
     type: mapVariableTypeEnum(id, role),
-    mandatory: true, // How shall we handle this? The value for elimination may differ in the jsonstat2-response depending on if all values are seleccted or not...
+    mandatory: true, // Default value
     values,
+    codeLists,
   };
 
+  mapDimensionExtension(dimension.extension, variable);
+
   return variable;
 }
 
+/**
+ * Maps the extension object of a dimension from a JSON-stat 2.0 response to a Variable object.
+ *
+ * @param extension - The extension object from the JSON-stat 2.0 response.
+ * @param variable - The Variable object to which the extension should be added.
+ */
+function mapDimensionExtension(extension: any, variable: Variable): void {
```

Comment:
I would really like it if we don't have 'any' if we can avoid it, is there a type you can narrow this down to later? So, set it to unknown first, then you would need to type it before you can use it later.
```suggestion
function mapDimensionExtension(extension: unknown, variable: Variable): void {
```